metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -142,23 +156,67 @@ public ObjectsImporter(
         this.taxonomyEventHandler = taxonomyEventHandler;
     }
 
-    public Study importStudy(String accession) throws Exception {
-        IMPORT_LOGGER.log(Level.INFO, "Importing study " + accession);
+    public Project importProject(String accession) throws Exception {
+        LOGGER.info("Importing project " + accession);
+        ProjectType projectType = retrieveType(accession, sraProjectXmlParser);
+        if (projectType == null) { return null; }
+        Project project = convertProject(projectType);
+
+        String studyAccession = projectType.getIDENTIFIERS().getSECONDARYIDArray(0).getStringValue();
+        StudyType studyType = retrieveType(studyAccession, sraStudyXmlParser);
+        if (studyType == null) {
+            throw new RuntimeException("All projects shold have a study ID");
+        }
+        Study study = convertStudy(studyType);
+        project = extractAnalysisFromProject(projectType, project, study);
```

Comment:
If I understand this correctly (from the code & from the PR description), the implementation you made is versatile so that the analyses can be extracted both from the project and from the study. This is convenient but also kind of leads to code/logic duplication.

An alternative would be, for the EVA use case, to always start from a project a go strictly from there: a project imports _only_ study, which in turn is responsible for importing _all other_ entities it requires. For the EGA case, I'm not so sure how this would work. Maybe we could investigate the EGA use case in a separate ticket, to see what is the reason for the so much different import process. (Maybe we could just start always start from project in both use cases?)

Reply:
I created  EVA-2012 for cleaning the relationships between the entities, and we can try to simplify the logic there.